django/db/models/fields/__init__.py

```diff
@@ -1522,8 +1522,11 @@ def get_internal_type(self):
     def to_python(self, value):
         if value is None:
             return value
+        context = decimal.Context(prec=self.max_digits)
+        if isinstance(value, float):
+            return context.create_decimal_from_float(value)
         try:
-            return decimal.Decimal(value)
+            return decimal.Decimal(value, context=context)
```

Comment:
Is `context` doing anything here? I don't see any test failures with it removed and I tried adding this test: `self.assertEqual(f.to_python('3.1415926535897'), Decimal('3.142'))` but it fails.